components/database.py

```diff
@@ -206,116 +239,153 @@ def get_all_past_mounted_modules():
     '''
         Get the fixed AY/Sem and quota of a mounted module
     '''
-    sql_command = "SELECT acadYearAndSem, quota FROM moduleMounted " +\
-                  "WHERE moduleCode=%s ORDER BY acadYearAndSem ASC"
-    DB_CURSOR.execute(sql_command, (code, ))
-    return DB_CURSOR.fetchall()
+    try:
+        sql_command = "SELECT acadYearAndSem, quota FROM moduleMounted " +\
+                      "WHERE moduleCode=%s ORDER BY acadYearAndSem ASC"
+        DB_CURSOR.execute(sql_command, (code, ))
+        return DB_CURSOR.fetchall()
+    except psycopg2.Error:
+        CONNECTION.rollback()
+        return []
 
 
 def get_tenta_mounting_and_quota(code):
     '''
         Get the tentative AY/Sem and quota of a mounted module
     '''
-    sql_command = "SELECT acadYearAndSem, quota FROM moduleMountTentative " +\
-                  "WHERE moduleCode=%s ORDER BY acadYearAndSem ASC"
-    DB_CURSOR.execute(sql_command, (code, ))
-    return DB_CURSOR.fetchall()
+    try:
+        sql_command = "SELECT acadYearAndSem, quota FROM moduleMountTentative " +\
+                      "WHERE moduleCode=%s ORDER BY acadYearAndSem ASC"
+        DB_CURSOR.execute(sql_command, (code, ))
+        return DB_CURSOR.fetchall()
+    except psycopg2.Error:
+        CONNECTION.rollback()
+        return []
 
 
 def get_mounting_of_target_fixed_ay_sem(code, ay_sem):
     '''
         Get the mounting status of a module in a target fixed AY/Sem
     '''
-    sql_command = "SELECT COUNT(*) FROM moduleMounted " +\
-                  "WHERE moduleCode=%s AND acadYearAndSem=%s"
-    DB_CURSOR.execute(sql_command, (code, ay_sem))
-    result = DB_CURSOR.fetchone()
-    return result[0] == 1    # True == Mounted, False == Not Mounted
-
+    try:
+        sql_command = "SELECT COUNT(*) FROM moduleMounted " +\
+                      "WHERE moduleCode=%s AND acadYearAndSem=%s"
+        DB_CURSOR.execute(sql_command, (code, ay_sem))
+        result = DB_CURSOR.fetchone()
+        return result[0] == 1    # True == Mounted, False == Not Mounted
+    except psycopg2.Error:
+        CONNECTION.rollback()
+        return []
 
 def get_mounting_of_target_tenta_ay_sem(code, ay_sem):
     '''
         Get the mounting status of a module in a target tentative AY/Sem
     '''
-    sql_command = "SELECT COUNT(*) FROM moduleMountTentative " +\
-                  "WHERE moduleCode=%s AND acadYearAndSem=%s"
-    DB_CURSOR.execute(sql_command, (code, ay_sem))
-    result = DB_CURSOR.fetchone()
-    return result[0] == 1    # True == Mounted, False == Not Mounted
-
+    try:
+        sql_command = "SELECT COUNT(*) FROM moduleMountTentative " +\
+                      "WHERE moduleCode=%s AND acadYearAndSem=%s"
+        DB_CURSOR.execute(sql_command, (code, ay_sem))
+        result = DB_CURSOR.fetchone()
+        return result[0] == 1    # True == Mounted, False == Not Mounted
+    except psycopg2.Error:
+        CONNECTION.rollback()
+        return []
```

Comment:
Similarly,the try block returns a boolean value, so the except clause should return a boolean value as well instead of an empty list.